tests/Boost_Tests/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+project(Boost_Tests)
+set(CMAKE_CXX_STANDARD 11)
+add_executable(${PROJECT_NAME}
+        main.cpp
+        )
+
+# Specific to Clare's Mac - temporary for testing...
+if (DEFINED ENV{USER})
+    if ("$ENV{USER}" STREQUAL "clare")
+        set(BOOST_ROOT /Users/clare/Documents/develop/tools/boost/boost_1_71_0/
+                CACHE FILEPATH "" FORCE)
+    endif ()
+endif ()
+
+# unit_test_framework
+find_package(Boost 1.1.0 COMPONENTS REQUIRED)
```

Comment:
Should this be Boost `1.71.0` instead of `1.1.0`?

Reply:
Thanks... This is an intentional temporary version number...  :-(

(When feeling fairly desperate to try and get this code building on various CI systems, we kept dialling back the version number, to match what was available e.g. via apt-get, each time we got a build failure. Eventually we get fed up of this, and set it to the lowest possible version..)

Some time soon, I'll work out when Boost added the features in `boost::unit_test::test_observer` that we are using - and then I'll set the version to the actual lowest version that we can support...

Reply:
The webpage on the test observer is available in the documentation of [Boost 1.62.0](https://www.boost.org/doc/libs/1_62_0/libs/test/doc/html/boost/unit_test/test_observer.html) but not in [Boost 1.61.0](https://www.boost.org/doc/libs/1_61_0/libs/test/doc/html/index.html). So maybe `1.62.0`?

Reply:
Fabulous - thank you very much!

Reply:
1.61.0 and 1.62 worked, so I narrowed it down:

```CMake
# /Users/clare/Downloads/boost_1_57_0 - no member named 'p_file_name' in 'boost::unit_test::test_unit'
# /Users/clare/Downloads/boost_1_58_0 - no member named 'p_file_name' in 'boost::unit_test::test_unit'
# /Users/clare/Downloads/boost_1_59_0 - error: no member named 'begin' in 'boost::unit_test::readonly_property<boost::unit_test::basic_cstring<const char> >'; did you mean to use '->' instead of '.'?
# /Users/clare/Downloads/boost_1_60_0 - works!!!
# /Users/clare/Downloads/boost_1_61_0 - works!!!
# /Users/clare/Downloads/boost_1_62_0 - works!!!
```

1.60 is the earliest - and I'll put that in code